text/0000-export-to-oci.md

```diff
@@ -0,0 +1,213 @@
+- IF any of the values are present THEN
+  - It will create a folder name `<image>` located at the path defined `<layout>` or `CNB_LAYOUT_DIR`
+  - In parallel of exporting the Image to the Daemon it will save the final Image in [OCI Image Layout](https://github.com/opencontainers/image-spec/blob/main/image-layout.md) format in the directory created before. The image will be saved using **uncompressed** layers
+  - It will calculate the digest of the manifest of the compressed layers and write that value into the report.toml file
+  - It will update the report.toml file with all the tags and require information to verify the image once it is pushed into a registry
+- OTHERWISE it will behave as it is right now
+
+#### `report.toml` (TOML)
+
+The new information to be  added into the `report.toml` file can be summarize as follows:
+
+```toml
+[export]
+[[export.oci]]
+digest = "<image digest>"
+manifest-size = "<manifest size in bytes>"
+compression-algorithm = "<used by gcr>"
+library-url = "https://github.com/google/go-containerregistry/"
+library-language="go"
+```
+Where:
+- **If** the app image was exported using the `-layer` flag, the export section will be added to the report
+  - `digest` MUST contain the image digest calculated based on compressed layers
+  - `manifest-size` MUST contain the manifest size in bytes
+  - `compression-algorithm` MUST contain the information of the algorithm used by GCR to compute the digest
+  - `library-url` MUST contain the url of the library used to export the image
+  - `library-language` MUST contain the client's library programming language used to export the image
+
+The main idea of this new section in the `report.toml` file is to provide the information of the **expected** digest of the image if it is exported to a registry using a particular implementation. This information can be used by other tools (like publish) to complete the operation and verify the consistency of the image.
+
+# Migration
+[migration]: #migration
+
+<!--
+This section should document breaks to public API and breaks in compatibility due to this RFC's proposed changes. In addition, it should document the proposed steps that one would need to take to work through these changes. Care should be give to include all applicable personas, such as platform developers, buildpack developers, buildpack users and consumers of buildpack images.
+-->
+# Drawbacks
+[drawbacks]: #drawbacks
+
+- We could increase the disk space if we not managed the duplication of saving the layers on disk. Currently the Cache implementation (used when daemon is ON) saved the layers tarballs on disk, because the current proposal is exporting the whole image on disk it will also require more space to save the layers for the OCI format in the `blobs` folder.
+
+# Alternatives
+[alternatives]: #alternatives
+
+## Redesign the current Cache
```

Comment:
```suggestion
## Redesign the current Launch Cache
```

Reply:
I'd be in favor of this solution (if it's possible, I haven't looked into all the details!). It would be nice not to effectively triplicate every daemon image (1 in daemon storage, 1 in launch cache, 1 in oci layout dir 😭 ).